plugins/module_utils/token_manager.py

```
@@ -0,0 +1,117 @@
+            "grant_type": "password",
+            "client_id": self.client_id,
+            "username": self.username,
+            "password": self.password,
+        })
+
+        response.raise_for_status()
+
+        # Store token data with expiration timestamps
+        self.token_data = {
+            "access_token": {
+                "token": token_data["access_token"],
```

Ruff (0.8.2)

Undefined name `token_data`

(F821)